### PR TITLE
[react-native] Add hover props to PressableProps

### PR DESCRIPTION
**Pull request by Saadnajmi**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested 

====

This is a followup to my now (ancient feeling) change that exposed new props on the Pressable component in React Native. This adds it to the types.

My change won't make it to React Native till RN 0.68, but I've backported it to older React Native macOS and React Native Windows releases. 
Relevant PRs:

RN Core change (0.68+): https://github.com/facebook/react-native/pull/32405

RN macOS and Windows changes + back ports: 

(RN macOS - 0.64) - https://github.com/microsoft/react-native-macos/pull/923
(macOS skipped RN 0.65, and went straight to 0.66)
(RN macOS - 0.66) - https://github.com/microsoft/react-native-macos/pull/884

(RN Windows - 0.64) - https://github.com/microsoft/react-native-windows/pull/9350
(RN Windows - 0.65) - https://github.com/microsoft/react-native-windows/pull/9349
(RN Windows - 0.66) - https://github.com/microsoft/react-native-windows/pull/9348
(RN Windows - 0.67) - https://github.com/microsoft/react-native-windows/pull/9347

### CODE REVIEW

**Comment by Saadnajmi** [comment]
@lunaleaps I spoke to @kelset who mentioned I should also open this PR against the RN repo. Any opposition to doing so? And should I only do that after this one merges?

**Comment by lunaleaps** [comment]
Thanks @Saadnajmi! Yes since this is a fix, let's have make dual PRs, the order doesn't matter.

**Comment by Saadnajmi** [comment]
> Thanks @Saadnajmi! Yes since this is a fix, let's have make dual PRs, the order doesn't matter. 

Thanks! I'm also kinda curious how these props with the new pointer API but I guess I'll find out eventually =)

File: types/react-native/v0.68/index.d.ts [hunk 1]
```diff
@@ -39,6 +39,7 @@
 //                 Pedro Hernández <https://github.com/phvillegas>
 //                 Sebastian Silbermann <https://github.com/eps1lon>
 //                 Zihan Chen <https://github.com/ZihanChen-MSFT>
+//                 Saad Najmi <https://github.com/saadnajmi>
```
Comment (lunaleaps) [inline types/react-native/v0.68/index.d.ts:42]: Did you mean to add this to the latest releases as well?

Comment (Saadnajmi) [reply types/react-native/v0.68/index.d.ts:42]: Good catch, thanks!

**Review by eps1lon — APPROVED** [review]

**Pull request merged**